jerry-core/ecma/base/ecma-helpers-external-pointers.c

```diff
@@ -176,49 +188,40 @@ ecma_delete_native_pointer_property (ecma_object_t *obj_p, /**< object to delete
 
   ecma_property_value_t *value_p = ECMA_PROPERTY_VALUE_PTR (property_p);
 
+  if (value_p->value == JMEM_CP_NULL)
+  {
+    return false;
+  }
+
   ecma_native_pointer_t *native_pointer_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_native_pointer_t,
                                                                              value_p->value);
-  ecma_native_pointer_t *prev_p = NULL;
-
   JERRY_ASSERT (native_pointer_p != NULL);
 
-  while (native_pointer_p != NULL)
+  ecma_native_pointer_t *prev_p = NULL;
+
+  do
   {
     if (native_pointer_p->info_p == info_p)
     {
       if (prev_p == NULL)
       {
-        if (native_pointer_p->next_p == NULL)
-        {
-          /* Only one native pointer property exists, so the property can be deleted as well. */
-          ecma_op_general_object_delete (obj_p, name_p, false);
```

Comment:
Is it ok to not remove the property storing native ptrs?

Reply:
That is intentional. Instead of creating deleted entires (holes), we can reuse the first entry if later a new native data is added.